Yoast/wordpress-seo
# Lingo 1488 caps not detected in keyphrases with a period

## Context


*

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Improves tokenizing text into sentences by adding a new condition that next sentence should be preceded by a whitespace except in Japanese.
* [yoastseo] Moves Japanese Sentence Tokenizer out of `SentenceTokenizer.js`.
* [shopify-seo] Fixes a bug where focus keyphrase in uppercase that contains a period wouldn't get recognized in the text, e.g. "ASP.NET".
* Fixes a bug where focus keyphrase in uppercase that contains a period wouldn't get recognized in the text, e.g. "ASP.NET".

## Relevant technical choices:

* Moved Japanese specific logic that's used to split Japanese text into sentences into its own tokenizer. 
  * This improves the separation of concerns between the tokenizer for latin character texts and Japanese character texts.
  * However, since we need it to rely on the standard tokenization rules, we made it extend the original tokenizer. 

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

**Test in WordPress**
* Create a post with at least 100 words length
* Set `ASP.NET` as the focus keyphrase
* Add these words in the text: `ASP.NET`, `ASP.net`, `asp.NET` and `asp.net`
* Confirm that the focus keyphrase is found 4 times in keyphrase density assessment
* Change the focus keyphrase to `"ASP.NET"`
* Confirm that the focus keyphrase is still found 4 times in keyphrase density assessment

**Test in Shopify**
* Create a product with at least 100 words length
* Set `ASP.NET` as the focus keyphrase
* Add these words in the text: `ASP.NET`, `ASP.net`, `asp.NET` and `asp.net`
* Confirm that the focus keyphrase is found 4 times in keyphrase density assessment
* Change the focus keyphrase to `"ASP.NET"`
* Confirm that the focus keyphrase is still found 4 times in keyphrase density assessment

### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

*

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Other environments


## Documentation


## Quality assurance


Fixes https://yoast.atlassian.net/browse/LINGO-1488
